eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -276,29 +276,52 @@ private void clusterSubmittedVariants(List<? extends SubmittedVariantEntity> sub
                                                               SubmittedVariantEntity.class);
         BulkOperations dbsnpBulkOperations = mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED,
                                                                    DbsnpSubmittedVariantEntity.class);
+        BulkOperations bulkHistoryOperations = mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED,
+                SubmittedVariantOperationEntity.class);
+        BulkOperations dbsnpBulkHistoryOperations = mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED,
+                DbsnpSubmittedVariantOperationEntity.class);
         long numUpdates = 0;
         long numDbsnpUpdates = 0;
         for (SubmittedVariantEntity submittedVariantEntity : submittedVariantEntities) {
             if (submittedVariantEntity.getClusteredVariantAccession() != null) {
                 // no need to update the rs of a submitted variant that already has the correct rs
                 continue;
             }
-            Query query = query(where("_id").is(submittedVariantEntity.getId()));
-            Update update = new Update();
-            update.set(RS_KEY, getClusteredVariantAccession(submittedVariantEntity));
+
+            long rsid = getClusteredVariantAccession(submittedVariantEntity);
+            // Query to update the RSid in submittedVariantEntity
+            Query rsQuery = query(where("_id").is(submittedVariantEntity.getId()));
+            Update rsUpdate = new Update();
+            rsUpdate.set(RS_KEY, rsid);
+
+            // Query to create the update operation history
+            SubmittedVariantOperationEntity update_operation = new SubmittedVariantOperationEntity();
+            SubmittedVariantInactiveEntity inactiveEntity = new SubmittedVariantInactiveEntity(submittedVariantEntity);
+            update_operation.fill(
+                    EventType.UPDATED,
+                    null,
+                    rsid,
```

Comment:
this accession should be the ssId

Reply:
let me explain a bit more. in submittedVariantOperationEntity, if you merge an ss2 into another ss1, you would say accessionIdOrigin=ss2, accessionIdDestiny=ss1, (in other words, ss2 merged into ss1). If the operation is not a MERGE, but an UPDATE (because we are updating a single field of the SS), then accessionIdDestiny does not apply, thus we leave it as null.

this way, we can ask "which operations happened to ss2?" by querying submittedVariantOperationEntity by accession=ss2 in mongo (the "accession" field in mongo is called "accessionIdOrigin" in java. this naming is a bit of a shame https://github.com/EBIvariation/accession-commons/blob/master/accession-commons-mongodb/src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/mongodb/document/EventDocument.java#L69)

Reply:
If I understand correctly:
for an operation:
 - The eventType is what actually happen: here it is an update
 - The accessionIdOrigin is what it happening it to: here it is the ssid
 - The accessionIdDestiny is only relevant in the event of a Merge ??
 - reason is a free text explanation 
 - inactiveEntities is the state of all relevant entities before the change: here it is the whole SubmittedVariant record

My only concern here is we're not recording the new RS id anywhere; except in the free text but that is not searchable.